src/GovUk.Education.ExploreEducationStatistics.Admin/Services/Public.Data/DataSetService.cs

```diff
@@ -60,6 +62,28 @@ public class DataSetService(
             });
     }
 
+    public async Task<Either<ActionResult, DataSetSummaryViewModel>> GetDataSet(
+        Guid dataSetId,
+        CancellationToken cancellationToken = default)
+    {
+        return await CheckDataSetExists(dataSetId, cancellationToken)
+            .OnSuccessCombineWith(dataSet => CheckPublicationExists(dataSet.PublicationId, cancellationToken))
+            .OnSuccess(async combinedDataSetAndPublication =>
+            {
+                (var dataSet, var publication) = combinedDataSetAndPublication;
```

Comment:
You can use `var (dataSet, publication) = combinedDataSetAndPublication` here and same below with `var (dataSet, releaseFilesByFileId) = combinedDataSetAndReleaseFiles`

Reply:
good shout